prober/icmp.go

```diff
@@ -65,14 +65,21 @@ func getICMPSequence() uint16 {
 
 func ProbeICMP(ctx context.Context, target string, module config.Module, registry *prometheus.Registry, logger log.Logger) (success bool) {
 	var (
-		socket      net.PacketConn
 		requestType icmp.Type
 		replyType   icmp.Type
+		icmpConn    *icmp.PacketConn
+		v4RawConn   *ipv4.RawConn
+		hopLimit    float64 = 0
```

Comment:
The = 0 is redundant, however it'd be cleaner to set it to -1 here.

Reply:
It can be either -1 or 0. I just need to set it to -1 if SetControlMessage fails or 0 if it works. As I don't use the "else" when I check SetControlMessage , I opted to use 0 by default and set it to -1 inside the existing block.

What would be the better option?

Reply:
A separate bool would be cleaner I think, rather than overloading this variable.

Reply:
It was the first approach. Updated.